R/db-odbc-mssql.R

```diff
@@ -100,3 +100,45 @@
   )
   DBI::dbExecute(con, sql)
 }
+
+#' @export
+`db_compute.Microsoft SQL Server` <- function(con,
```

Comment:
This seems to be identical to `db_compute.DBIConnection`?

Reply:
I fixed `temporary` to TRUE in `db_save_query`

Reply:
In `db_compute.DBIConnection`, `db_save_query` has `temporary = temporary`

Reply:
Don't we want that?

Reply:
This change was based on my initial understanding of `compute()`.  I'll remove that customization

Reply:
`db_compute.Microsoft SQL Server` should have some lines added to prefix # to the table name. See my comment above dated 18 Sep. Not sure why/when it got dropped.

This is my original version from the comment:

```r
#' @export
`db_compute.Microsoft SQL Server` <- function(con, table, sql, temporary=TRUE, unique_indexes=list(), indexes=list(), ...)
{
    # check that name has prefixed '##' if temporary
    if(temporary && substr(table, 1, 1) != "#")
        table <- paste0("##", table)

    if(!is.list(indexes))
        indexes <- as.list(indexes)

    if(!is.list(unique_indexes))
        unique_indexes <- as.list(unique_indexes)

    db_save_query(con, sql, table, temporary=temporary)
    db_create_indexes(con, table, unique_indexes, unique=TRUE)
    db_create_indexes(con, table, indexes, unique=FALSE)
    table
}
```

Reply:
That's correct.   If the user passes `compute()`, meaning w/o a name, then that name will generate a "##" warning.  So the prefix needs to be done silently.  Additionally, the change needs to happen in `db_compute()` because the `table` variable will now have a different name.

Reply:
@Hong-Revo - Can you check my latest commit to make sure I restored back your changes.  I re-tested the changes against a MS SQL connection and everything worked fine